pkg/operator/ceph/object/user/controller.go

```diff
@@ -257,6 +349,13 @@ func (r *ReconcileObjectStoreUser) reconcile(request reconcile.Request) (reconci
 		return reconcileResponse, *cephObjectStoreUser, err
 	}
 
+	// update status of referenced secrets only after the rgw user has reconciled
+	if referencedSecrets != nil {
+		// update even when no secrets are referenced as this could be a transition
+		// from explicit keys -> automatic secret generation
+		r.updateKeyStatus(request.NamespacedName, referencedSecrets)
```

Comment:
I'm trying to figure out if the `if referencedSecrets != nil` is going to lead to a corner case where referenced secrets are disabled by removing the section and leading to it being nil again. From the comment here, it seems like maybe we want to just always call `updateKeyStatus()`, but I'm not certain

Reply:
I believe you are correct and the status should always be updated.  The integration tests do check the status field when changing the explicit keys but not in the transition case.  Removing the automatic secret deletion transition required a lot of integration test changes and this probably got lost in the shuffle.

Reply:
I take that back -- There is integration test coverage that shows the status field being cleaned up when removing explicit keys: https://github.com/rook/rook/pull/15359/files#diff-baae21f06d33f385aa56617fbdfbda8d5d4fbac79722f7270d6c3696d0a3f8e0R507-R513

Reply:
I think it is safe to remove the conditional as `referencedSecrets` should only be `nil` when there was an error that would cause a return. https://github.com/rook/rook/pull/15359/files#diff-4399505e3e48def2feaa495cba5133401361caddd0f494399172bab3acb15a64R311

Reply:
Done.